OricExplorer/Classes/OricTape.cs

```diff
@@ -226,67 +212,76 @@ private ProgramHeader ReadProgramHeader()
 
-            // Get the programs end address
-            highByte = m_bTapeBuffer[m_ui32BufferIdx];
-            lowByte = m_bTapeBuffer[m_ui32BufferIdx + 1];
+                    // Get the programs end address
+                    highByte = m_bTapeBuffer[m_ui32BufferIdx];
+                    lowByte = m_bTapeBuffer[m_ui32BufferIdx + 1];
 
-            programHeader.ui16EndAddress = Convert.ToUInt16(lowByte + (highByte * 256));
+                    programHeader.ui16EndAddress = Convert.ToUInt16(lowByte + (highByte * 256));
 
-            m_ui32BufferIdx += 2;
+                    m_ui32BufferIdx += 2;
 
-            // Get the programs start address
-            highByte = m_bTapeBuffer[m_ui32BufferIdx];
-            lowByte = m_bTapeBuffer[m_ui32BufferIdx + 1];
+                    // Get the programs start address
+                    highByte = m_bTapeBuffer[m_ui32BufferIdx];
+                    lowByte = m_bTapeBuffer[m_ui32BufferIdx + 1];
 
-            programHeader.ui16StartAddress = Convert.ToUInt16(lowByte + (highByte * 256));
+                    programHeader.ui16StartAddress = Convert.ToUInt16(lowByte + (highByte * 256));
 
-            if(bProgramFormat == 0x00)
-                programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.BasicProgram;
-            else
-            {
-                if(programHeader.ui16StartAddress == 0xA000)
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.HiresScreen;
-                else if(programHeader.ui16StartAddress == 0xB500)
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.CharacterSet;
-                else if(programHeader.ui16StartAddress.In((ushort)0xBB80, (ushort)0xBBA8))
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.TextScreen;
-                else
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.BinaryFile;
-            }
+                    if (bProgramFormat == 0x00)
+                        programHeader.Format = OricProgram.ProgramFormat.BasicProgram;
+                    else
+                    {
+                        if (programHeader.ui16StartAddress == 0xA000)
+                            programHeader.Format = OricProgram.ProgramFormat.HiresScreen;
+                        else if (programHeader.ui16StartAddress == 0xB500)
+                            programHeader.Format = OricProgram.ProgramFormat.CharacterSet;
+                        else if (programHeader.ui16StartAddress.In((ushort)0xBB80, (ushort)0xBBA8))
+                            programHeader.Format = OricProgram.ProgramFormat.TextScreen;
+                        else
+                            programHeader.Format = OricProgram.ProgramFormat.BinaryFile;
+                    }
 
-            m_ui32BufferIdx += 3;
+                    m_ui32BufferIdx += 3;
 
-            while(m_bTapeBuffer[m_ui32BufferIdx] != 0)
```

Comment:
This loop for reading the program filename in the .TAP header must be strengthened as it can cause an exception in cases where the NUL terminator ('\0') is not found in the header (endless loop causing overflow of m_bTapeBuffer).
Proposed change:  given that the program name is at most 16 chars, the loop should be rewritten as follows with another counter to limit max iterations to 16:

   int iProgNameLength = 0;
    while( (m_bTapeBuffer[m_ui32BufferIdx] != 0) && (iProgNameLength < 16) ) {
                programHeader.strProgramName += Convert.ToChar(m_bTapeBuffer[m_ui32BufferIdx]);
                m_ui32BufferIdx++;
		iProgNameLength++;
   }

Reply:
While different from my initial proposal, the modification to the original loop condition implemented here works equally well.